### ISSUE TITLE
[javascript] try/catch unnecessary block

### ISSUE THREAD

**Issue by dah-fari7009** [pmd/pmd#4054]
**Affects PMD Version:** 
6.36.0

**Rule:**
UnnecessaryBlock
<https://pmd.github.io/latest/pmd_rules_ecmascript_codestyle.html#unnecessaryblock>

**Description:**
PMD reports an unnecessary block for the try/catch, but there's no reason why this should be the case. Further, looking into the generate AST by PMD, it seems to be an issue with handling async/await keywords. In fact, without:
![image](https://user-images.githubusercontent.com/35702512/179899775-379babe3-30c6-4beb-a588-38fb9ef6e0b2.png)

and with it, it shows EmptyStatement instead:
![image](https://user-images.githubusercontent.com/35702512/179899922-bca3501a-dc79-425f-91b5-7649e11fbd1a.png)

**Code Sample demonstrating the issue:**

```
async function run(){
    try{
        await client.connect()
        console.log("Successfully connected to the database")
        var server = app.listen(8081, function (){
            var host = server.address().hostname
            var port = server.address().port
            console.log("Example app running at https://%s:%s", host, port)
        })
    }
    catch(err){
        console.log(err)
        await client.close()

    }
}
```

**Expected outcome:**

PMD reports a violation at line 2 for the try block, but that seems to be a false positive.

**Running PMD through:** *[Codacy]*

**Comment by jsotuyod** [comment]
@dah-fari7009 PMD relies on Mozilla Rhino for Ecmascript parsing. The version of PMD you are using is quite old (over a year old actually), and is missing the latest Rhino version. The latest PMD release (6.47.0) is using the very latest version of Rhino. If using it you still face this issue, please, feel free to reopen this ticket.

**Comment by dah-fari7009** [comment]
Hi,

Thank you for the follow-up. The version of PMD used by Codacy is 6.36.0 but those AST were generated using the latest version of PMD from the website (6.47.0 with javafx sdk 18.0.1), . Sorry about the confusion
